### PR TITLE
[GH-7349] Using Product-Icon at Site-level

### PR DESCRIPTION
**Pull request by manujgrover71**
#### Summary

This PR addresses the issue of standardizing product icons across the documentation site. 

It introduces a centralized definition for all existing product icons, allowing them to be referenced site-wide using a short code. By defining icons in a single location, the PR streamlines the process of updating and managing icons, reducing redundancy and improving maintainability across the docs site.

Some icons were inconsistently referenced using different names, so this PR consolidates them under a single reference. Below is a summary of the updates:

| Reference Name 1 | Reference Name 2 | Final Reference Name |
| -- | -- | -- |
| `gear-icon` | `gear` | `gear` |
| `save-icon` | `bookmark-icon` | `save-icon` |
| `plus-icon` | `plus` | `plus` |
| `globe-icon` | `globe ` | `globe` |
| `lock-icon` | `lock` | `lock` |
| `links-icon` | `copy-link-icon` | `copy-link-icon` |
| `more-icon-vertical` | `vertical-3-dots` | `more-icon-vertical` |
| `more-actions-icon` | `more-icon` | `more-icon` |
| `priority-icon` | `message-priority-icon` | `message-priority-icon` |
| `acknowledged-icon` | `acknowledge-button` | `acknowledge-button` |
| `attachment-icon` | `attachments-icon` | `attachments-icon` |
| `product-menu` | `product-list` | `product-list` |

Additionally, `channel-files-icon.svg` was found to be identical to `file-text-outline_F09EE.svg`. All instances of `channel-files-icon.svg` were updated to use `file-text-outline_F09EE.svg`, and the redundant file was removed.

During testing, I observed that `restore-edit` icon wasn't changing colour based on the theme (light / dark) even though the `theme-icon` class was associated with the same. It's currently live on production as well.

image

[Doc Link](https://docs.mattermost.com/collaborate/send-messages.html#restore-a-previous-version-of-an-edited-message)

#### Ticket Link

Fixes: https://github.com/mattermost/docs/issues/7349

### CODE REVIEW

**Comment by cwarnermm** [comment]
@manujgrover71 - Thank you so much for this most excellent docs PR! I really appreciate your help with this lift. Following a quick spot-check, I noticed that some product icon definitions at the page level are missing from the ``conf.py`` file. For example: https://github.com/mattermost/docs/blob/master/source/collaborate/format-messages.rst

Would you be open to performing a second pass to identify and migrate remaining instances where we're defining product icons at the page level instead of system-wide?

**Comment by manujgrover71** [comment]
Absolutely, @cwarnermm. I just double-checked and confirmed that there are still some places where these icons are defined at the page level. I'll update my PR with the necessary changes accordingly.

**Comment by manujgrover71** [comment]
@cwarnermm

According to the original issue, product icons should have the theme-icon class attached to them. However, on the [Collect performance metrics page](http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/7374/scale/collect-performance-metrics.html#usage), the Plus Icon doesn't have the theme-icon class applied.

I believe this icon should include the theme-icon class. If that's correct, should I make the necessary changes in this PR?

**Comment by cwarnermm** [comment]
Great catch, @manujgrover71! Your interpretation is correct: all product icons should include the ``theme-icon class``. Any omissions are an oversight. Please feel free to include fixes to these oversights in this docs PR.

File: source/conf.py [hunk 1]
```diff
@@ -3725,6 +3725,127 @@ def setup(_: Sphinx):
...
+    :class: theme-icon
+.. |away| image:: /images/away.png
+    :alt: Away availability status icon in Mattermost.
+    :class: theme-icon
+.. |dnd| image:: /images/dnd.png
+    :alt: Do Not Disturb availability status icon in Mattermost.
+    :class: theme-icon
+.. |offline| image:: /images/offline.png
+    :alt: Offline availability status icon in Mattermost.
+    :class: theme-icon
```
Comment (cwarnermm) [inline source/conf.py:3848]:
```suggestion
.. |offline| image:: /images/offline.png
    :alt: Offline availability status icon in Mattermost.
```

File: source/conf.py [hunk 2]
```diff
@@ -3725,6 +3725,127 @@ def setup(_: Sphinx):
...
+.. |online| image:: /images/online.png
+    :alt: Online availability status icon in Mattermost.
+    :class: theme-icon
+.. |away| image:: /images/away.png
+    :alt: Away availability status icon in Mattermost.
+    :class: theme-icon
+.. |dnd| image:: /images/dnd.png
+    :alt: Do Not Disturb availability status icon in Mattermost.
+    :class: theme-icon
```
Comment (cwarnermm) [inline source/conf.py:3845]:
```suggestion
.. |dnd| image:: /images/dnd.png
    :alt: Do Not Disturb availability status icon in Mattermost.
```

File: source/conf.py [hunk 3]
```diff
@@ -3725,6 +3725,127 @@ def setup(_: Sphinx):
...
+.. |servers-icon| image:: /images/server-variant_E81F.svg
+    :alt: Access server connection options using the Servers icon.
+    :class: theme-icon
+.. |online| image:: /images/online.png
+    :alt: Online availability status icon in Mattermost.
+    :class: theme-icon
+.. |away| image:: /images/away.png
+    :alt: Away availability status icon in Mattermost.
+    :class: theme-icon
```
Comment (cwarnermm) [inline source/conf.py:3842]:
```suggestion
.. |away| image:: /images/away.png
    :alt: Away availability status icon in Mattermost.
```

File: source/conf.py [hunk 4]
```diff
@@ -3725,6 +3725,127 @@ def setup(_: Sphinx):
...
+.. |edit-on-github| image:: /images/edit-on-github.png
+    :alt: Contribute to Mattermost documentation by selecting the Edit option located in the top right corner of any documentation page.
+    :class: theme-icon
+.. |servers-icon| image:: /images/server-variant_E81F.svg
+    :alt: Access server connection options using the Servers icon.
+    :class: theme-icon
+.. |online| image:: /images/online.png
+    :alt: Online availability status icon in Mattermost.
+    :class: theme-icon
```
Comment (cwarnermm) [inline source/conf.py:3839]:
```suggestion
.. |online| image:: /images/online.png
    :alt: Online availability status icon in Mattermost.
```

**Review by cwarnermm — APPROVED** [review]
Thank you, @manujgrover71, for this incredible lift!

**Pull request merged**